receiver/awscontainerinsightreceiver/receiver.go

```diff
@@ -39,6 +41,7 @@ type awsContainerInsightReceiver struct {
 	cancel            context.CancelFunc
 	cadvisor          metricsProvider
 	k8sapiserver      metricsProvider
+	k8swindows        metricsProvider
```

Comment:
my two cents - k8swindows and cadvisor are **the same**, no reason to not just populate cadvisor with two different impls, one impl for windows and one impl for linux (actually using cadvisor).

Then you don't need special code in the receiver for k8swindows.GetMetrics()

I am open to renaming it, kubeletProvider or something

Reply:
Addressed.
I have removed `k8swindows` from this struct. Renamed `cadvisor` field to  `containerMetricProvider` and reused it for both cadvisor and k8sWindows implementation.